[next] scene graph refactoring (#1093)
threlte/threlte: apps/docs/src/content/learn/advanced/migration-guides.mdx

This will pass the new callback props mentioned in the previous section down the tree of components.

### Attach API & Trait Components

The signature and heuristic of the attach API of the `<T>` component has
changed. The trait components `<HierarchicalObject>` and `<SceneGraphObject>`
have been removed.

#### `attach` Function Signature

```svelte
<!-- Threlte 7 -->
<T.Mesh
  attach={(parent, self) => {
    console.log('attaching', parent, self)
    return () => {
      console.log('detaching', parent, self)
    }
  }}
/>

<!-- Threlte 8 -->
<T.Mesh
  attach={({ ref, parent, parentObject3D }) => {
    console.log('attaching', ref, parent, parentObject3D)
    return () => {
      console.log('detaching', ref, parent, parentObject3D)
    }
  }}
/>
```

#### `attach={false}`

> When passing `false` as the attach prop, the component will not be attached to

Reviewer: ```suggestion
If `false` is passed to the `attach` prop, the component will not be automatically attached to
```

the parent object. This is useful if you want to attach the component manually.

```svelte
<!-- "Dangling" component -->
<T
  is={mesh}
  attach={false}
/>
```

#### `attach={object3D}`

> When passing an object3D as the attach prop, the component will be attached to
> the provided object, essentially acting as a portal.

Reviewer: ```suggestion
If an object3D instance is passed to the `attach` prop, the component will be attached to the instance, essentially acting as a portal.
```
